File: JavaScript/Valid-Anagram-Easy.js
// Problem Link: https://leetcode.com/problems/valid-anagram/description/

// Space Complexity: O(n)
//  Time Complexity: O(nˆ3)

// Performace:
//     RunTime:   366   ms  ( better then 05,06% of users of Java )
//      Memory:   55.82 MB  ( better then 06,37% of users of Java ) 


/**
 * @param {string} s
 * @param {string} t
 * @return {boolean}
 */
var isAnagram = function(s, t) {
  if(s.length != t.length) return false 

  for (let i = 0; i < s.length; i++) {
    t = t.replace(s.charAt(i), "")
  }

  if(t.length == 0) return true
  return false
};

File: Java/Valid-Anagram-Easy.java
// Problem Link: https://leetcode.com/problems/valid-anagram/description/

// Space Complexity: O(n)
//  Time Complexity: O(n)

// Performace:
//     RunTime:   33.00 ms  ( better then 05,10% of users of Java )
//      Memory:   44.22 MB  ( better then 55,59% of users of Java ) 

package Java;
import java.util.HashMap;

class Solution {
  public boolean isAnagram(String s, String t) {
    if(s.length() != t.length()) return false;

    HashMap<Character, Integer> charCountS = new HashMap<Character, Integer>();
    HashMap<Character, Integer> charCountT = new HashMap<Character, Integer>();

    for(int i=0; i<s.length(); i++) {
      if (charCountS.containsKey(s.charAt(i))) {
        charCountS.put(s.charAt(i), charCountS.get(s.charAt(i))+1);
      } else {
        charCountS.put(s.charAt(i), 1);
      }
      
      if (charCountT.containsKey(t.charAt(i))) {
        charCountT.put(t.charAt(i), charCountT.get(t.charAt(i))+1);
      } else {
        charCountT.put(t.charAt(i), 1);
      }
    }

    if (charCountS.equals(charCountT)) {
      return true;
    }
    return false;
  }
}